Library/Formula/fstar.rb

```diff
@@ -0,0 +1,69 @@
+class Fstar < Formula
+  desc "F*, an ML-like language with a type system for program verification"
+  homepage "http://fstar.gforge.inria.fr/"
+  url "https://github.com/FStarLang/FStar.git", :tag => "v0.9.0",
+                                                :revision => "f24457213f4007302873f6d42eb2cf8094fa45d0"
+  head "https://github.com/FStarLang/FStar.git", :branch => "master"
+
+  depends_on "mono"
+  depends_on "fsharp"
+  depends_on "ocaml"
+  depends_on "homebrew/science/z3"
```

Comment:
Can't have mandatory cross-tap dependencies in the core. You'll either need to make this optional, vendor it into `libexec/"vendor"` or such, or submit this to the Science tap.